// @crowdin/crowdin-api-client/out/fields/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Fields = void 0;
const core_1 = require("../core");
class Fields extends core_1.CrowdinApi {
    /**
     * @param options optional parameters for the request
     * @see https://developer.crowdin.com/enterprise/api/v2/#operation/api.fields.getMany
     */
    listFields(options) {
        let url = `${this.url}/fields`;
        url = this.addQueryParam(url, 'search', options === null || options === void 0 ? void 0 : options.search);
        url = this.addQueryParam(url, 'entity', options === null || options === void 0 ? void 0 : options.entity);
        url = this.addQueryParam(url, 'type', options === null || options === void 0 ? void 0 : options.type);
        return this.getList(url, options === null || options === void 0 ? void 0 : options.limit, options === null || options === void 0 ? void 0 : options.offset);
    }
    /**
     * @param request request body
     * @see https://developer.crowdin.com/enterprise/api/v2/#operation/api.fields.post
     */
    addField(request) {
        const url = `${this.url}/fields`;
        return this.post(url, request, this.defaultConfig());
    }
    /**
     * @param fieldId field identifier
     * @see https://developer.crowdin.com/enterprise/api/v2/#operation/api.fields.get
     */
    getField(fieldId) {
        const url = `${this.url}/fields/${fieldId}`;
        return this.get(url, this.defaultConfig());
    }
    /**
     * @param fieldId field identifier
     * @see https://developer.crowdin.com/api/v2/#operation/api.fields.delete
     */
    deleteField(fieldId) {
        const url = `${this.url}/fields/${fieldId}`;
        return this.delete(url, this.defaultConfig());
    }
    /**
     * @param fieldId field identifier
     * @param request request body
     * @see https://developer.crowdin.com/api/v2/#operation/api.fields.patch
     */
    editField(fieldId, request) {
        const url = `${this.url}/fields/${fieldId}`;
        return this.patch(url, request, this.defaultConfig());
    }
}
exports.Fields = Fields;
